pkg/util/util.go

```diff
@@ -35,8 +35,28 @@ var (
 		"registry-1.docker.io":        true,
 		"registry.connect.redhat.com": true,
 	}
+
+	savedCommonDockerRegistries map[string]bool
 )
 
+// UpdateCommonRegistries add additional registries to common registries.
+func UpdateCommonRegistries(additionalRegistries []string) {
```

Comment:
This function will eventually forget about the original common registries and keep on adding to the list.

I think we should avoid saving such a list in utils package in a variable. It is error prone to keep such a singleton util.
I would suggest that in the GetImageURN function we can just pass the list of additional registries (or the whole cluster object and then get the annotation) and replace the common registry if present in the additional list on top of the common registries.

Reply:
Sounds good, I can make it as another parameter of GetImageURN